.github/workflows/main.yml

```diff
@@ -0,0 +1,54 @@
+name: CI
+
+on:
+  push:
+
+jobs:
+  python-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+          cache: 'pip'
+      - name: Install dependencies
+        working-directory: ./camera
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r ./requirements.txt
+      - name: Run tests
+        working-directory: ./camera
+        run: make test
+
+  app-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'npm'
+          cache-dependency-path: app/human-detector-app/package-lock.json
+      - name: Install dependencies
+        working-directory: ./app/human-detector-app
+        run: npm install
+      - name: Run tests
+        working-directory: ./app/human-detector-app
+        run: npm test
+  
+  backend-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'npm'
+          cache-dependency-path: server/backend/package-lock.json
+      - name: Install dependencies
+        working-directory: ./server/backend
+        run: npm install
```

Comment:
Honestly not entirely sure what difference it makes, but there's an [`npm ci`](https://docs.npmjs.com/cli/v8/commands/npm-ci) command that's apparently meant to be used in place of `npm install` for CI

Reply:
ah, looks like it clears cache and reinstalls everything. If we did that, we'd want to disable caching of dependencies.